MIGRATION_GUIDE.md

```diff
@@ -11,17 +11,24 @@ across different versions.
 
 ### *(new feature)* connection resources
 
-Added a new resources for managing connections. We decided to split connection into two separate resources based on whether the connection is primary or a replica (secondary). i.e.:
+Added a new resources for managing connections. We decided to split connection into two separate resources based on whether the connection is a primary or replicated (secondary). i.e.:
 
-- `snowflake_connection` is used as primary connection, with ability to enable failover to other accounts.
-- `snowflake_secondary_connection` is used as replica (secondary) connection.
+- `snowflake_primary_connection` is used to manage primary connection, with ability to enable failover to other accounts.
+- `snowflake_secondary_connection` is used to manage replicated (secondary) connection.
 
-In order to promote secondary_connection to primary, resources need to be migrated (check [resource migration](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/technical-documentation/resource_migration.md)) or re-created and imported using the following SQL statements on Snowflake Worksheet:
+In order to promote `snowflake_secondary_connection` to `snowflake_primary_connection`, resources need to be removed from the state, altered manually using:
+```
+ALTER CONNECTION <name> PRIMARY;
+```
+and then imported again, now as `snowflake_primary_connection`.
 
+In order to demote `snowflake_primary_connection` back to `snowflake_secondary_connection`, resources need to be removed from the state, re-created manually using:
 ```
 CREATE CONNECTION <name> AS REPLICA OF <organization_name>.<account_name>.<connection_name>;
```

Comment:
You are right, but you may have a number of references pointing to this connection that may prevent you from running `terraform apply` without it, then you have to go through this process. I guess this could be clarified at the top.

Reply:
We can destroy resources with a target like `terraform destroy --target aws_instance.demo_vm_1`

Reply:
But still, I'm not sure how the TF apply reacts to those dependencies; with import, I feel safer because I know that the state is in the correct form **before** the actual tf apply and not will be fixed during it.

Reply:
Should be addressed in the next pr (if needed).

Reply:
To me, this is optional; this way is good, too.